### PR TITLE
DT-2929: ✨ Add in email addresses to results

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt [hunk 1]
```diff
@@ -145,3 +150,36 @@ class CaseloadAlreadyExistsException(message: String?) :
     return CaseloadAlreadyExistsException(message)
   }
 }
+
+fun UserPersonDetail.toUserSummaryWithEmail() = UserSummaryWithEmail(
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt:154]: wasn't a great fan of having a `Transformers` class that will just get bigger and bigger, so moved the summary stuff here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserSummaryWithEmail.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.nomisuserrolesapi.data
+
+import io.swagger.v3.oas.annotations.media.Schema
+
+@Schema(description = "Summary User Information with Email Address")
+data class UserSummaryWithEmail(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserSummaryWithEmail.kt:6]: I think perhaps a better approach would be to just have a single UserSummary class and add email to main the `/users` endpoint.

In fact I think we were going to do this at some point any way so "find users" doesn't rely on Auth.

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserSummaryWithEmail.kt (hunk 2 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/data/UserSummaryWithEmail.kt:6]: agreed, will pick up in next PR

**Pull request merged**